projects/vdk-plugins/vdk-impala/tests/functional/test_impala_connection.py

```diff
@@ -42,3 +45,26 @@ def test_execute_query(self) -> None:
         )
         cli_assert_equal(0, actual_rs)
         assert "GOOG" in actual_rs.output
+
+    @patch.dict(
+        os.environ,
+        {
+            VDK_DB_DEFAULT_TYPE: "IMPALA",
+            VDK_IMPALA_HOST: "localhost",
+            VDK_IMPALA_PORT: "21050",
+            VDK_IMPALA_SYNC_DDL: "True",
+            VDK_IMPALA_QUERY_POOL: "testing_grounds",
+        },
+    )
+    @patch("vdk.plugin.impala.impala_plugin.DecorationCursor.execute")
```

Comment:
Hmm. I'd prefer we do not use patching in functional tests. 

Functional tests often serves as a proxy/example for users using given feature. 

We have a pretty powerful plugin system that's perfect for way to "mock" in a way an end user would

You can create a a plugin - similar to https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-test-utils/src/vdk/plugin/test_utils/util_plugins.py#L102  QueriesRecorderPlugin which can record SQL statements. And now you can verify below that 'sync_ddl" was recorded by the plugin

Reply:
Unfortunately the approach seen in the linked place doesn't work - the SYNC_DDL and REQUEST_POOL decoration queries are executed through the decoration cursor, and as such they do not get ran through the decoration hook - I verified this. I'm open to other suggestions but I see no clear way of plugging into the execution of those queries, except through some sort amendment of the decoration implementation.

Personally, I don't see an issue with the patching that happens here, as no behaviour is altered, and only the call_args_list is checked for the necessary queries. It's possible my understanding of the mocking module is wrong, though.

Reply:
In this case the problem is not that behaviour is altered. The problem is that the functional test relies on internal implementation classes instead of relying on public contracts (even if it's in verification step only)

If in the future some developer decides to rename DecorationCursor or change the method signature of DecorationCursor#execute (add new parameter) -  it would break this test despite that the actual logic being tested is not broken. 
Also I do think that functional test could be even used as an example to show users how things can be done. They need to mimic what users can do. 

But if it is not possible. I guess we can live with it. I might have been wrong it's a good idea to be a functional test this.